### PR TITLE
Migrate injection example to use linear-handles feature.

### PR DESCRIPTION
**Pull request by wildarch**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

Tracking issue: #1854

### CODE REVIEW

File: sdk/rust/oak/src/io/sender.rs [hunk 1]
```diff
@@ -43,12 +43,14 @@ impl<T: Encodable> SenderExt<T> for Sender<T> {
     fn send(&self, t: &T) -> Result<(), OakError> {
         let message = t.encode()?;
         crate::channel_write(&self.handle, &message.bytes, &message.handles)?;
+        drop_handles(message);
```
Comment (rbehjati) [inline sdk/rust/oak/src/io/sender.rs:46]: I am not sure why the handles need to be dropped. Are not they borrowed or moved?

**Review by rbehjati — APPROVED** [review]
LGTM, thanks.

File: sdk/rust/oak/src/io/sender.rs (hunk 1 above)
Comment (wildarch) [reply sdk/rust/oak/src/io/sender.rs:46]: During encoding we extract the raw handle values from the message, and replace them with `INVALID_HANDLE`: https://github.com/project-oak/oak/blob/main/oak_io/src/handle.rs#L202.

When this message is dropped, all handle values are invalid, so no handles are dropped. Instead we need to manually drop the raw handle values here. I hope that makes sense?

Comment (rbehjati) [reply sdk/rust/oak/src/io/sender.rs:46]: Yes. Thanks. I had overlooked a `clone` :)

**Review by rbehjati — APPROVED** [review]

**Pull request merged**